core/src/main/java/com/sopt/core/designsystem/component/timetable/NoostakEditableTimeTable.kt

```diff
@@ -0,0 +1,156 @@
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.sopt.core.designsystem.theme.NoostakTheme
+import com.sopt.core.extension.noRippleClickable
+import com.sopt.domain.entity.AvailableTimeEntity
+import com.sopt.domain.entity.TimeEntity
+import com.sopt.domain.entity.TimeTableEntity
+
+@Composable
+fun NoostakEditableTimeTable(
+    data: TimeTableEntity,
+    modifier: Modifier = Modifier,
+    onSelectionChange: (List<TimeEntity>) -> Unit
+) {
+    val days = data.timeEntity.size
+    val timeSlots = calculateTimeSlots(data.startTime, data.endTime)
+    val selectedCells = remember { mutableStateListOf<Pair<Int, Int>>() } // Row, Column 저장
+
+    LazyVerticalGrid(
+        modifier = modifier
+            .border(
+                width = 1.dp,
+                color = NoostakTheme.colors.gray200,
+                shape = RoundedCornerShape(8.dp)
+            ),
+        columns = GridCells.Fixed(days + 1)
+    ) {
+        items((days + 1) * (timeSlots + 1)) { index ->
+            val rowIndex = index / (days + 1)
+            val columnIndex = index % (days + 1)
+
+            val cellType = determineCellType(rowIndex, columnIndex)
+            val isSelected = selectedCells.contains(rowIndex to columnIndex)
+            val backgroundColor =
+                getEditableBackgroundColor(cellType, rowIndex, columnIndex, isSelected)
+            val text = getCellText(cellType, rowIndex, columnIndex, data)
+
+            NoostakEditableTimeTableBox(
+                index = index,
+                days = days,
+                timeSlots = timeSlots,
+                backgroundColor = backgroundColor,
+                text = text,
+                onClick = {
+                    if (cellType == CellType.Data) {
+                        val cell = rowIndex to columnIndex
+                        if (selectedCells.contains(cell)) {
+                            selectedCells.remove(cell)
+                        } else {
+                            selectedCells.add(cell)
+                        }
+                        onSelectionChange(
+                            selectedCells.toTimeEntities(
+                                data.startTime,
+                                data.timeEntity
+                            )
+                        )
+                    }
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun getEditableBackgroundColor(
+    cellType: CellType,
+    rowIndex: Int,
+    columnIndex: Int,
+    isSelected: Boolean
+): Color {
+    return when (cellType) {
+        CellType.Blank, CellType.DateHeader, CellType.TimeHeader -> Color.Transparent
+        CellType.Data -> if (isSelected) NoostakTheme.colors.blue400 else Color.Transparent
+    }
+}
```

Comment:
P1: 여기 rowIndex, columnIndex 사용 안하는거면 지우셔도 될 것 같습니다!
+) return 없애고 표현식 쓰기~!

```suggestion
@Composable
fun getEditableBackgroundColor(
    cellType: CellType,
    isSelected: Boolean
): Color = when (cellType) {
    CellType.Blank, CellType.DateHeader, CellType.TimeHeader -> Color.Transparent
    CellType.Data -> if (isSelected) NoostakTheme.colors.blue400 else Color.Transparent
}
```